src/Storages/MergeTree/MergeTreeIndexVectorSimilarity.cpp

```diff
@@ -382,7 +384,7 @@ void MergeTreeIndexAggregatorVectorSimilarity::update(const Block & block, size_
     if (index->size() + rows > std::numeric_limits<UInt32>::max())
         throw Exception(ErrorCodes::INCORRECT_DATA, "Size of vector similarity index would exceed 4 billion entries");
 
-    DataTypePtr data_type = block.getDataTypes()[0];
```

Comment:
This was the problem.

Reply:
Is the data type in `block` possible to be different? Why not to use `block.getByName(index_column_name).type`?

Reply:
Changed to `block.getByName(index_column_name).type`.

In theory, `block` and `index_sample_block` could have different data types but the other skipping indexes also don't check, so I think it is only a theoretical risk.